allowed.py

```diff
@@ -367,10 +368,19 @@ def get_constructs(last_unit: int) -> tuple:
     )
 
 
-def check_tree(tree: ast.AST, constructs: tuple, source: list) -> list:
+def check_tree(
+    tree: ast.AST,
+    constructs: tuple,
+    source: list,
+    from_ipynb: bool,
+    line_cell_map: dict,
+) -> list:
     """Check if tree only uses allowed constructs."""
     language, options, imports, functions, methods = constructs
     errors = []
+    get_line = (
+        lambda l: f"cell_{line_cell_map[l][0]}:{line_cell_map[l][1]}" if from_ipynb else l
+    )
```

Comment:
I confess I'm not a big fan of lambdas, because they obscure what the inputs mean. An auxiliary function `get_line`, before `check_tree`, that takes a dict and a line and returns the latter if the dict is empty, would be more  readable IMHO.

Reply:
Yeah that's fair enough,  I can agree with that, I will break `get_line` out into its own function definition and I guess having the doc string will add to the readability too.